device.go

```diff
@@ -90,10 +103,35 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
 		return syscall.ENOSPC
 	}
 
-	w, err := io.Copy(b, image)
+	ssz, err := b.SectorSize()
 	if err != nil {
-		log.Errorf("failed to write image data to device %v: %v",
+		log.Errorf("failed to read sector size of block device %s: %v",
 			inactivePartition, err)
+		return err
+	}
+
+	buf := make([]byte, ssz)
```

Comment:
`CopyBuffer` definitely looks interesting. Will take a closer look and test it out.